_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -5267,6 +5286,19 @@ mfxStatus ConfigureExecuteParams(
             case MFX_EXTBUFF_VPP_DEINTERLACING:
             case MFX_EXTBUFF_VPP_DI:
             {
+                try
+                {
+                    mfxStatus support_status = CheckFilterSupport(GetPlatformGen(hwType),
+                        DEINTERLACING,
+                        MFXFourCC2VPPSupportTableFourCC.at(videoParam.vpp.In.FourCC),
```

Comment:
ok, looks like you have this code at each switch, maybe we can consolidate this try-catch before ```switch``` ?

Reply:
Yes, we can convert MFXFourCC to format which is used in CheckFilterSupport function and write it to some variable before _for loop_ and consolidate all loop in try-catch

Reply:
Why do you need try-catch for all loop? You only need to place ```.at()``` calls inside try-catch block

Reply:
`VPPSupportTableFourCC inFourCC = MFXFourCC2VPPSupportTableFourCC.at(videoParam.vpp.In.FourCC);`
I want do this before loop. If I close `try` I can't use `inFourCC` in loop. If we catch exception we don't need to run loop

Reply:
Workaround is simple :)

```
VPPSupportTableFourCC inFourCC, outFourCC;
try
{
    inFourCC  = MFXFourCC2VPPSupportTableFourCC.at(videoParam.vpp.In.FourCC);
    outFourCC = MFXFourCC2VPPSupportTableFourCC.at(videoParam.vpp.Out.FourCC);
}
catch (const std::out_of_range & ex)
{
    MFX_RETURN(MFX_ERR_INVALID_VIDEO_PARAM); // didn't find input or output FourCC in VPPSupportTableFourCC
}

for (mfxU32 j = 0; j < pipelineList.size(); j += 1)
{
...
}
```

Reply:
Yes, it is also possible, looks prettier, thanks

Reply:
Can't do so because I added `VPPExtBuf2VPPSupportTableFilter` map to convert filters and it is used in many cases in loop `VPPExtBuf2VPPSupportTableFilter.at(curFilterId)`